ksql-metastore/src/main/java/io/confluent/ksql/metastore/ReadonlyMetaStore.java

```diff
@@ -16,7 +16,6 @@
 
 import io.confluent.ksql.function.AggregateFunctionFactory;
 import io.confluent.ksql.function.KsqlAggregateFunction;
-import io.confluent.ksql.function.KsqlFunction;
```

Comment:
Not specific to the changes in this PR, but I think it would be better to implement this at the interface level by having a MetaStore interface with functions that don't support writing, and a MetaStoreWriter interface with the functions for writing. MetaStoreImpl can implement both. This way we fail at compile-time instead of run-time if we try to read when we shouldn't. Its also nicer for folks writing code to get feedback about misuse of the metastore early rather than getting an exception when testing the code out.

Reply:
Sure, like I've one for `FunctionRegistry` you mean?  Yep, but one for another PR.

Reply:
> Sure, like I've one for `FunctionRegistry` you mean?

Yeah exactly - I reviewed the metastore changes first :D